src/client.js

```diff
@@ -12,37 +12,61 @@ function subKey(event, args) {
 }
 
 /**
- * @typedef {RPCClientOptions}
- * @extends {ClientOptions}
- * @prop {string} transport RPC transport. one of `ipc` or `websocket`
+ * @typedef {string} Snowflake A Twitter snowflake, except the epoch is 2015-01-01T00:00:00.000Z
+ * {@link https://discord.com/developers/docs/reference#snowflakes}
+ */
+
+/**
+ * @typedef {Object} BaseApplication
+ * @prop {string} name The application name
+ * @prop {Snowflake} application_id The application ID
+ */
+
+/**
+ * @typedef {Object} RPCClientOptions
+ * @prop {'ipc' | 'websocket'} transport RPC transport
+ * @prop {string} [origin] Origin - Used for `websocket` transport
  */
 
 /**
  * The main hub for interacting with Discord RPC
- * @extends {BaseClient}
+ * @extends {EventEmitter}
  */
 class RPCClient extends EventEmitter {
   /**
-   * @param {RPCClientOptions} [options] Options for the client.
+   * @param {RPCClientOptions} [options] Options for the client
    * You must provide a transport
    */
   constructor(options = {}) {
     super();
 
+    /**
+     * Options for this client
+     * @type {RPCClientOptions}
+     */
     this.options = options;
 
+    /**
+     * The access token for the connected user
+     * @type {?string}
+     */
     this.accessToken = null;
+
+    /**
+     * The client ID
+     * @type {?Snowflake}
+     */
     this.clientId = null;
 
     /**
      * Application used in this client
-     * @type {?ClientApplication}
+     * @type {?Object<string, *>}
      */
     this.application = null;
 
     /**
-     * User used in this application
-     * @type {?User}
+     * User user in this application
```

Comment:
Not sure this is wanted
```suggestion
     * User used in this application.
```